modules/github-bots/sdk/github.go

```diff
@@ -433,6 +434,86 @@ func (c GitHubClient) ListArtifactsFunc(ctx context.Context, wr *github.Workflow
 	return nil
 }
 
+// GetChangedFiles uses the git package to get a map of the files changed between two branches.
+func (c GitHubClient) GetChangedFiles(_ context.Context, repo *git.Repository, from, to string) (map[string]struct{}, error) {
+	if repo == nil {
+		return nil, fmt.Errorf("repository is nil")
+	}
+
+	parseRef := func(ref string) (*plumbing.Reference, error) {
+		if plumbing.IsHash(ref) {
+			return plumbing.NewHashReference(plumbing.ReferenceName(ref), plumbing.NewHash(ref)), nil
+		}
+		if !strings.HasPrefix(ref, "refs/") {
+			ref = "refs/heads/" + ref
+		}
+		return repo.Reference(plumbing.ReferenceName(ref), true)
+	}
+
+	fromRef, err := parseRef(from)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get reference for %s: %w", from, err)
+	}
+
+	toRef, err := parseRef(to)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get reference for %s: %w", to, err)
+	}
+
+	if fromRef == nil || toRef == nil {
+		return nil, fmt.Errorf("one or both references are nil")
+	}
+
+	fromCommit, err := repo.CommitObject(fromRef.Hash())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get commit for %s: %w", fromRef.Hash().String(), err)
+	}
+
+	toCommit, err := repo.CommitObject(toRef.Hash())
+	if err != nil {
+		return nil, fmt.Errorf("failed to get commit for %s: %w", toRef.Hash().String(), err)
+	}
+
+	if fromCommit == nil || toCommit == nil {
+		return nil, fmt.Errorf("one or both commits are nil")
```

Comment:
nit: include the `fromCommit` and `toCommit` in the error message so its clear which one is nil or not